### ISSUE TITLE
Helping demean() Fail Gracefully - Make 'maxiter' Accessible from feols() 

### ISSUE THREAD

**Issue by b-knight** [py-econometrics/pyfixest#539]
I've encountered instances when [demean_.demean()](https://github.com/py-econometrics/pyfixest/blob/master/pyfixest/estimation/demean_.py#L187-L267) runs endlessly - probably because I passed it a poorly structured data shape. I would like to be able to call a service that uses PyFixest, so calling the package on problematic data is a distinct possibility. 

To this end, it would be great if there were ways to help the demean() method fail gracefully. Having some sort of mechanism for timing out could be handy, but the simplest approach that I can think of off hand is to make the 'maxiter' argument accessible from the feols() method in a manner similar to how the 'fixef_tol' argument is accessible to the user.

**Comment by s3alfisc** [comment]
Yes, I agree - it would be great to add a `fixef_iter` function argument to `pyfixest.estimation.feols` and `pyfixest.estimation.fepois` that would allow users to set the maximum number of iterations for the demeaning algo. 

Overall, this should be straightforward enough: 

- We'd have to add a function argument `fixef_iter` to both functions & add docstrings. 
- We'd then have to propagate the argument to the `FixestMulti` class (just as `fixef_tol`) and assign it as an attribute. 
- For objects of class `Feols` and `Feiv`, from `FixestMulti`, we'd directly pass it to `demean_model`, where we finally can pass it to `demean`.
- For objects of class `Feiv`, we'd pass the argument to `Fepois`, from which we can pass it to the internal calls to [demean](https://github.com/py-econometrics/pyfixest/blob/a476ec541c5093a1666aad284c4f7eb42631483c/pyfixest/estimation/fepois_.py#L206).
